yt_dlp/extractor/jditemvideo.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+import json
+import random
+import time
+
+from .common import InfoExtractor
+from ..utils import determine_ext, ExtractorError, traverse_obj
+
+
+class JdItemVideoIE(InfoExtractor):
+    _VALID_URL = r"https://.+.jd.[a-z\.]{2,9}/(?P<id>\d{6,16}).html"
+
+    IE_NAME = 'jd-video'
+    IE_DESC = 'jd-video extractor'
+    _NETRC_MACHINE = False
+
+    _JD_API_VIDEO_CALLBACK_URL = 'https://cd.jd.com/tencent/video_v3?callback=jQuery{rand}&vid={video_id}&type=1&from=1&appid=24&_={timestamp}'
+
+    _TESTS = [
+        {
+            'url': 'https://npcitem.jd.hk/100030101538.html',
+            'info_dict': {
+                "id": "100030101538",
+                "ext": "mp4",
+                "title": "ipad 2021第九代",
+                "description": "【AppleiPad】Apple苹果 iPad 第9代 10.2英寸平板电脑 2021款 ipad9（64GB WLAN版/A13芯片/1200万像素/iPadOS）深空灰色【行情 报价 价格 评测】-京东",
+                "size": 10251794,
+                "width": 1280,
+                "height": 1280,
+                "duration": 56,
+                "thumbnail": "https://jvod.300hu.com/img/2022/130871763/1/img7.jpg",
+                "url": "https://jvod.300hu.com/vod/product/6e02e2d8-98bc-491d-80a1-448ae5ea1c38/c6ef7b9b14ef4b9ca7e4cebda5b7684c.mp4?source=2&h265=h265/18799/a797504bd6f947dfbf6fdb96acfbb55f.mp4",
```

Comment:
Single quotes for all of the strings in the tests

Let the test script generate the test data for you:
Remove all fields except `id` and `ext` and run the test, e.g. `python test/test_download.py TestDownload.test_JDItemVideo`
It will output an info_dict for you to copy/paste